apis/duck/v1alpha1/condition_set.go

```diff
@@ -278,3 +280,24 @@ func (r conditionsImpl) InitializeCondition(t ConditionType) {
 		})
 	}
 }
+
+// getConditions uses reflection to return the field on the status called "Conditions".
+func (r conditionsImpl) getConditions() Conditions {
+	conditionsField := reflect.Indirect(reflect.ValueOf(r.status)).FieldByName("Conditions")
```

Comment:
Can you assert that the struct contains a "Conditions" field in the constructor (i.e. `Manage()`)?

Reply:
Also, check that `conditionsField.Interface()` can be cast to `Conditions` in the constructor?

Reply:
> Can you assert that the struct contains a "Conditions" field in the constructor (i.e. Manage())?

I could use the `duck` type testing that was added I think. I was having issues trying that because it was not clear how to use a generic `{}interface` but I can try again.

>Also, check that conditionsField.Interface() can be cast to Conditions in the constructor?

I do the check on line 289.

Reply:
>> Also, check that conditionsField.Interface() can be cast to Conditions in the constructor?
>
>I do the check on line 289.

If you do the check in the constructor before storing `r.status`, you do the check once per construction. If you do the check in `getCondition`/`setCondition`, you do the check once per method call.